resolwe/flow/views.py

```diff
@@ -135,23 +135,35 @@ class ResolweCreateDataModelMixin(ResolweCreateModelMixin):
     """
     def create(self, request, *args, **kwargs):
         collections = request.data.get('collections', [])
-        if len(collections) != 1:
-            return Response({'collections': 'Exactly one id required on create.'},
-                            status=status.HTTP_400_BAD_REQUEST)
-
-        try:
-            collection = Collection.objects.get(pk=collections[0])
-        except Collection.DoesNotExist:
-            return Response({'collections': ['Invalid pk "{}" - object does not exist.'.format(collections[0])]},
-                            status=status.HTTP_400_BAD_REQUEST)
-
-        if not request.user.has_perm('add_collection', obj=collection):
-            if request.user.is_authenticated():
-                raise exceptions.PermissionDenied
-            else:
-                raise exceptions.NotFound
+        
+        for collection_id in collections:
```

Comment:
I agree that failing whole transaction is easiest solution. But do we want to return just first error or all of them?
We have to thing what to support and what not. If something won't be used much and is to complicated to think through all possibilities then I don't see a reason not to disable it.